src/main/java/com/edmunds/tools/databricks/maven/BaseDatabricksMojo.java

```diff
@@ -21,63 +21,89 @@
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
 /**
  * The base databricks mojo.
  */
 public abstract class BaseDatabricksMojo extends AbstractMojo {
 
+    private static final String DB_USER = "DB_USER";
+    private static final String DB_PASSWORD = "DB_PASSWORD";
+    private static final String DB_URL = "DB_URL";
+    private static final String DB_TOKEN = "DB_TOKEN";
+
     @Parameter(defaultValue = "${project}", readonly = true)
     protected MavenProject project;
 
     /**
-     * The environment name. Is used in freemarker templating for conditional job settings.e
+     * The environment name. Is used in freemarker templating for conditional job settings.
      */
-    @Parameter(property="environment")
+    @Parameter(property = "environment")
     protected String environment;
 
-    @Parameter(property="host")
+    @Parameter(property = "host")
     protected String host;
 
-    @Parameter(property="token")
+    @Parameter(property = "token")
     protected String token;
 
-    @Parameter(property="user")
+    @Parameter(property = "user")
     protected String user;
 
-    @Parameter(property="password")
+    @Parameter(property = "password")
     protected String password;
 
     @Parameter(defaultValue = "true", property = "validate")
     protected boolean validate;
 
 
-
     private DatabricksServiceFactory databricksServiceFactory;
 
     protected DatabricksServiceFactory getDatabricksServiceFactory() {
 
         if (databricksServiceFactory == null) {
-            //TODO temporary, will be replaced with logic to get environment variables
-            if (environment == null || host == null) {
-                throw new IllegalArgumentException("Must specify environment and host");
-            }
+
+            loadPropertiesFromSystemEnvironment();
+
             if (user != null && password != null) {
                 return DatabricksServiceFactory
-                    .Builder
-                    .createServiceFactoryWithUserPasswordAuthentication(user, password, host)
-                    .build();
+                        .Builder
+                        .createServiceFactoryWithUserPasswordAuthentication(user, password, host)
+                        .build();
             } else if (token != null) {
                 return DatabricksServiceFactory
-                    .Builder
-                    .createServiceFactoryWithTokenAuthentication(token, host)
-                    .build();
+                        .Builder
+                        .createServiceFactoryWithTokenAuthentication(token, host)
+                        .build();
             } else {
                 throw new IllegalArgumentException("Must either specify user/password or token!");
             }
         }
         return databricksServiceFactory;
     }
 
+    private void loadPropertiesFromSystemEnvironment() {
+        String envUser = System.getenv(DB_USER);
```

Comment:
so i am ok with this as it is. A gotya for users is that user password will always trump token access, How about add a comment to the token property above?

Reply:
That works for me.

Reply:
Done. Check again.